### PR TITLE
Use pretty user names for replies, fix typo, predictable !help ordering

### PR DESCRIPTION
**Pull request by lbatalha**

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 def help(cmd):
 	if cmd.args: # only reply on "!help"
 		return
-	commands = set(cmd.bot.commands.keys())
+	commands = cmd.bot.commands.keys()
```
Comment (raylu) [inline utils.py:25]: wait we call `commands.remove()` down below

Comment (lbatalha) [reply utils.py:25]: Its still a list, so no problem there?

Comment (raylu) [reply utils.py:25]: ain't no `list`
```py
>>> a = {}
>>> a.keys().remove()
AttributeError: 'dict_keys' object has no attribute 'remove'
```

Comment (lbatalha) [reply utils.py:25]: replaced with `del commands[name]`

Comment (raylu) [reply utils.py:25]:
```py
>>> a = {'cow': 'moo'}
>>> c = a.keys()
>>> del c['cow']
TypeError: 'dict_keys' object does not support item deletion
```

Comment (lbatalha) [reply utils.py]: converting to a list it is

**Pull request merged**